### DISCUSSION TITLE
How to set the default value of a field to value in it's relation?

### DISCUSSION THREAD

**Discussion by damafeez** [prisma/prisma#20125] [Q&A] [answered]
Thank you for all the good work on Prisma.

I have a model
```prisma
model Group {
  id      String  @id @default(cuid())
  name String
  private Boolean
}
```

And another model that references the circle
```prisma
model SmallerGroup {
  id      String      @id @default(cuid())
  name String
  private Boolean  @default(true)

  group   Group @relation(fields: [groupId], references: [id])
  groupId String
}
```

I want the private field on the smaller group to default to the private field on the larger/parent group, like so:

```prisma
model SmallerGroup {
  id      String      @id @default(cuid())
  name String
  private Boolean  @default(group.private) <---- this errors, how to do something like this?

  group   Group @relation(fields: [groupId], references: [id])
  groupId String
}
```

Such that when the small group is created without a `private` field, the default value is the `group.private`, I don't know if this is possible in Prisma (or in relational db languages, generally).

**Comment by nurul3101** [comment, marked as answer]
Hello @damafeez 👋 

You would need to handle this at the application level. The default value of a record cannot be dependent on the value of another record. Relational databases generally do not support this kind of default value referencing across tables. The concept of "default value" in SQL is quite primitive and limited to constant values.

When you create a `SmallerGroup`, you can first fetch the `Group` it's related to, get the value of private from that group, and use it as the default value for the `SmallerGroup`. This would be the most straightforward way to implement this.

It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 
Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by damafeez** [comment]
I thought as much. Still wanted to verify to ensure I wasn't missing something.

Thanks.
